src/routes/landing/components/Landing.js

```diff
@@ -25,12 +26,35 @@ export type PropsType = {
   theme: ThemeType,
   navigateToDashboard: (cityCode: string, language: string) => StoreActionType,
   fetchCities: () => StoreActionType
-}
+|}
+
+type StateType = {|
+  localizationLoaded: boolean
+|}
+
+class Landing extends React.Component<PropsType, StateType> {
+  constructor (props: PropsType) {
+    super(props)
+    this.state = { localizationLoaded: false }
+  }
+
+  componentWillMount () {
+    this.loadSelectedCity()
+  }
+
+  async loadSelectedCity () {
+    const { language, navigateToDashboard } = this.props
+    const localizationSettings = new LocalizationSettings()
+
+    const selectedCity = await localizationSettings.loadSelectedCity()
+
+    if (selectedCity) {
+      navigateToDashboard(selectedCity, language)
+    } else {
+      this.setState({localizationLoaded: true})
+    }
```

Comment:
I think this switch should be already done in the Navigator to avoid jumpy screens at the start of the app and not even navigate to the LandingPage first

Reply:
Done. It adds a bit more complexity and there is still kind of a jumpy screen, but should be ok.